src/main/java/org/ibp/api/java/impl/middleware/germplasm/GermplasmServiceImpl.java

```diff
@@ -207,24 +207,38 @@ public long countSearchGermplasm(final GermplasmSearchRequest germplasmSearchReq
 	}
 
 	@Override
-	public PedigreeDTO getPedigree(final Integer germplasmDbId, final String notation, final Boolean includeSiblings) {
+	public PedigreeDTO getPedigree(final String germplasmDbId, final String notation, final Boolean includeSiblings) {
 		final PedigreeDTO pedigreeDTO;
 		try {
-			pedigreeDTO = this.germplasmDataManager.getPedigree(germplasmDbId, notation, includeSiblings);
-			if (pedigreeDTO != null) {
-				pedigreeDTO.setPedigree(this.pedigreeService.getCrossExpansion(germplasmDbId, this.crossExpansionProperties));
+			final Optional<GermplasmDTO> germplasmDto = this.germplasmService.getGermplasmDTOByGUID(germplasmDbId);
+			if (germplasmDto.isPresent()) {
+				final int gid = Integer.valueOf(germplasmDto.get().getGid());
+				pedigreeDTO = this.germplasmService.getPedigree(gid, notation, includeSiblings);
+				if (pedigreeDTO != null) {
+					pedigreeDTO.setPedigree(this.pedigreeService.getCrossExpansion(gid, this.crossExpansionProperties));
+				}
+			} else {
+				throw new ApiRuntimeException("Invalid GermplasmDbId");
 			}
+
 		} catch (final MiddlewareQueryException e) {
 			throw new ApiRuntimeException("An error has occurred when trying to get the pedigree", e);
 		}
 		return pedigreeDTO;
 	}
 
 	@Override
-	public ProgenyDTO getProgeny(final Integer germplasmDbId) {
+	public ProgenyDTO getProgeny(final String germplasmDbId) {
 		final ProgenyDTO progenyDTO;
 		try {
-			progenyDTO = this.germplasmDataManager.getProgeny(germplasmDbId);
+			final Optional<GermplasmDTO> germplasmDTO = this.germplasmService.getGermplasmDTOByGUID(germplasmDbId);
```

Comment:
1. Call this.germplasmValidator.validateGermplasmGUID and throw the exception right away. You can create a `validateGUID` method:
```
this.errors = new MapBindingResult(new HashMap<String, String>(), Integer.class.getName());
		this.germplasmValidator.validateGermplasmGUID(errors, germplasmDbId);
		if (this.errors.hasErrors()) {
			throw new ResourceNotFoundException(this.errors.getAllErrors().get(0));
		}
```
2. Remove `isPresent` checking since you've already validated the GUID exists
3. Remove throwing of ApiRuntimeExeption("invalid germplasmDbId) 
4. Remove catching MiddlewareQueryException